test/unit/adapters/http.js

```diff
@@ -789,6 +789,56 @@ describe('supports http with nodejs', function () {
     });
   });
 
+  it('should re-evaluate proxy on redirect when proxy set via env var', function (done) {
+    process.env.http_proxy = 'http://localhost:4000'
+    process.env.no_proxy = 'localhost:4000'
+
+    var proxyUseCount = 0;
+
+    server = http.createServer(function (req, res) {
+      res.setHeader('Location', 'http://localhost:4000/redirected');
+      res.statusCode = 302;
+      res.end();
+    }).listen(4444, function () {
+      proxy = http.createServer(function (request, response) {
+        var parsed = url.parse(request.url);
+        if (parsed.pathname === '/redirected') {
```

Comment:
I (somewhat hackily) re-used the proxy as both a proxy and the redirected endpoint. That saved me from starting another server to handle the redirection and having to deal with that extra server in `afterEach`. It would be cleaner, but that's for a single test. Let me know what you prefer.

Reply:
I think the single test is fine for now 👍